src/channel/rpc_0_6.ts

```diff
@@ -168,6 +169,22 @@ export class RpcChannel {
     return this.fetchEndpoint('starknet_getBlockWithTxs', { block_id });
   }
 
+  public async getL1MessageHash(txHash: BigNumberish) {
+    const txData = (await this.getTransactionByHash(txHash)) as any;
+    const { calldata, contract_address, entry_point_selector, nonce, version, type } = txData;
+    const hashedCallData = computeHashOnElements(calldata);
+
+    return computeHashOnElements([
+      useEncoded(type),
```

Comment:
I think it's not the right encoding. Use the same encoding used for calculation of transaction hash : `shortString.encodeShortString()`

Reply:
Done